test/cookbooks/openstack_test/recipes/hosts_provisioning.rb

```diff
@@ -1,18 +1,38 @@
 controller = []
 
+# Find the IP for the controller node
 unless Chef::Config[:solo]
+  public_ip = nil
+  local_ip = nil
+  # Search for any public IPs
   search(:node, 'recipes:osl-openstack\:\:controller',
          filter_result: { 'ip' => %w(cloud public_ipv4) }).each do |result|
-    controller << result['ip']
+    public_ip = result['ip']
+  end
+  # Search for any local IPs
+  search(:node, 'recipes:osl-openstack\:\:controller',
+         filter_result: { 'ip' => %w(cloud local_ipv4) }).each do |result|
+    local_ip = result['ip']
+  end
+  # Don't set if we don't find any
+  unless (public_ip.nil? && local_ip.nil?) || (public_ip.empty? && local_ip.empty?)
```

Comment:
If one is `nil` and one is `empty`, wouldn't this still pass?

Worse yet, will this crash on the right half of the `or` statement?

Reply:
I don't believe any of those conditions could happen. Also, this code is only used for testing using provisioning and not production so I wouldn't worry about it that much.